### DISCUSSION TITLE
What features do you want to see?

### DISCUSSION THREAD

**Discussion by DamianMorozov** [DamianMorozov/OpenTgResearcher#4] [Ideas]
Write your expectations for future versions.

**Comment by DamianMorozov** [comment]
How about storing the connection information in a local sqlite database?

**Comment by DamianMorozov** [comment]
This functions is done.

**Comment by ghost** [comment]
I tested the program and here are some thoughts. When the option to attach msg id to the filename is set to true, the download is going pretty fast, but when the option is set to false, the download is slowing down significantly. Also, after some time, the program waits quite a long period before resuming downloads.

It will be good if improvements are done in this area. A convenient feature is the ability to choose what to download, maybe adding some blacklists/whitelists based on file extension, so the users can better control the content they want to download.

**Reply by DamianMorozov** [reply]
It's nice to read the first reviews. How exactly do you see this setting in the program?

**Reply by ghost** [reply]
Using the telegram categories: video, audio and document. But if users blacklist the document category, they could miss some audios and videos send as documents (files). So an additional filter based on file extensions will do the job. User will whitelist audio, video categories and also .mp4 and .mkv file extensions. This will gather all needed files.

**Reply by DamianMorozov** [reply]
I will be adding a "customize filters" option soon.

**Comment by Windows55** [comment]
Thanks to the developer. I clicked on all the options and did not find the proxy server settings, are there any plans to add this feature?

**Reply by DamianMorozov** [reply]
Thank you for your ideas.
Here's a link to a poll of what features you need more.
https://t.me/TgDownloaderChat/6

**Reply by DamianMorozov** [reply]
Proxy configuration will appear in the next release.

**Reply by DamianMorozov** [reply]
Can you give me some specific proxy settings for the test?

**Comment by TooShyTo** [comment]
Some Telegram groups have subgroups or topics. It would be great if your downloader could recognize these topics and organize all the group's media into corresponding subfolders instead of placing everything in a single folder.

**Reply by TooShyTo** [reply]
qwq

**Comment by DamianMorozov** [comment]
Perhaps this feature would be better suited to the desktop version. I don't quite know how to implement it yet.

**Comment by DamianMorozov** [comment]
Already implemented.
